### PR TITLE
Several improvements to README.md files

### PR DESCRIPTION
**Pull request by jtattermusch**
-- get rid of places that mentions "Status beta"
-- update C# readme (add .NET core support and remove stuff no longer relevant)
-- add/improve readme.md for some subdirs

### CODE REVIEW

**Comment by jtattermusch** [comment]
CC @murgatroid99

**Comment by jtattermusch** [comment]
This is a documentation only change.

File: src/csharp/README.md [hunk 1]
```diff
@@ -4,10 +4,13 @@ gRPC C#
...
 A C# implementation of gRPC.
 
-Status
-------
+SUPPORTED PLATFORMS
+------------------
+
+- .NET Framework 4.5+ (Windows)
+- [.NET Core](https://dotnet.github.io/) on Linux, Windows and Mac OS X (starting from version 1.0.1)
+- Mono 4+ on Linux, Windows and Mac OS X
```
Comment (apolcyn) [inline src/csharp/README.md]: nit: I know tests of the .NET core target are under actual .NET core platform, but should we refer to more general netstandard platform here, in  case one wants to consume it from something that's not .NET core but netstandard1.5-compatible?

Comment (jtattermusch) [inline src/csharp/README.md]: Yeah, you're right. I'll update once I have a bit of time.

**Pull request merged**